### ISSUE TITLE
[release/2.12][BMG B70] Dynamo benchmark accuracy regression in PyTorch 2.12 release: 8 models fail_accuracy (passed in 2.11)

### ISSUE THREAD

**Issue by mengfei25** [intel/torch-xpu-ops#3457]
## Summary

8 dynamo benchmark models show `fail_accuracy` regression on **Intel BMG B70** GPU with **PyTorch 2.12 release** (test/xpu wheel). All 8 models **passed** in the PyTorch 2.11 baseline.

## Environment

- **GPU**: Intel BMG B70
- **Docker image**: `intelgpu/ubuntu-24.04-rolling:26.05`
- **PyTorch**: 2.12 (test/xpu wheel from `https://download.pytorch.org/whl/test/xpu`)
- **Baseline**: PyTorch 2.11 (stable/xpu wheel)

## Failing Models

| Suite | Data Type | Mode | Model | Failure Reason | Baseline |
|-------|-----------|------|-------|----------------|----------|
| huggingface | amp_fp16 | inference | AllenaiLongformerBase | dtype mismatch fp16→bf16 | pass |
| huggingface | amp_fp16 | inference | DistillGPT2 | dtype mismatch fp16→bf16 | pass |
| huggingface | amp_fp16 | inference | Qwen/Qwen3-0.6B | dtype mismatch fp16→bf16 | pass |
| huggingface | amp_fp16 | inference | google/gemma-2-2b | dtype mismatch fp16→bf16 | pass |
| huggingface | amp_fp16 | inference | openai/whisper-tiny | dtype mismatch fp16→bf16 | pass |
| timm_models | amp_bf16 | training | adv_inception_v3 | RMSE tolerance exceeded | pass |
| timm_models | amp_fp16 | training | fbnetv3_b | RMSE tolerance exceeded | pass |
| timm_models | bfloat16 | training | fbnetv3_b | RMSE tolerance exceeded | pass |

### Category 1: dtype mismatch (5 huggingface models, amp_fp16 inference)

All 5 huggingface models fail with the same error pattern:

```
E torch/_dynamo/utils.py:3380] dtype mismatch torch.float16, torch.bfloat16
E torch/_dynamo/utils.py:3357] Accuracy failed for key name logits
fail_accuracy
```

The inductor-compiled model outputs `torch.bfloat16` tensors when `torch.float16` is expected under `amp_fp16` mode. This suggests a regression in AMP autocasting or dtype handling on XPU for inference.

### Category 2: RMSE tolerance exceeded (3 timm models, training)

**adv_inception_v3** (amp_bf16 training):
```
RMSE (res-fp64): 0.01013, (ref-fp64): 0.00177 and shape=torch.Size([384])
Accuracy failed for key name Mixed_7c.branch3x3dbl_2.bn.running_mean
```

**fbnetv3_b** (amp_fp16 training):
```
RMSE (res-fp64): 0.00467, (ref-fp64): 0.00122 and shape=torch.Size([16])
Accuracy failed for key name blocks.0.0.bn1.bias.grad
```

**fbnetv3_b** (bfloat16 training):
```
RMSE (res-fp64): 0.05237, (ref-fp64): 0.01628 and shape=torch.Size([16])
Accuracy failed for key name bn1.bias.grad
```

fbnetv3_b fail_accuracy should be related with Triton, https://github.com/intel/intel-xpu-backend-for-triton/issues/6489

All 3 timm training failures involve **batch normalization** parameters (running_mean or bias.grad), suggesting a potential numerical regression in batch norm backward/update on XPU.

## Reproducer

```bash
docker run -it -e TZ=Asia/Shanghai --device=/dev/mem --device=/dev/dri --group-add video --privileged --shm-size=8g intelgpu/ubuntu-24.04-rolling:26.05 bash

curl -LsSf https://astral.sh/uv/install.sh | env UV_INSTALL_DIR="/usr/local/bin" sh
uv venv myvenv --python 3.12 --clear
source myvenv/bin/activate
uv pip install pip numpy 'setuptools<81' wheel

uv pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/test/xpu  # 2.12

pytorch_commit="$(python -c 'import torch; print(torch.version.git_version)')"
git clone https://github.com/pytorch/pytorch
cd pytorch
git checkout ${pytorch_commit}
rm -rf torch torch-xpu-ops
git clone https://github.com/intel/torch-xpu-ops
rsync -avz ./torch-xpu-ops/.ci/benchmarks/ ./benchmarks/dynamo/

uv pip install pip pandas psutil scipy requests
uv pip install -U numpy==1.26.4
uv pip install transformers==4.55.2 accelerate
uv pip install timm==1.0.19

# Example: reproduce dtype mismatch on DistillGPT2
python ./benchmarks/dynamo/torchbench.py --accuracy --amp --amp-dtype float16 --inference -d xpu -n 10 --only DistillGPT2 --backend=inductor --cold-start-latency --timeout 10800 --disable-cudagraphs

# Example: reproduce RMSE failure on fbnetv3_b
python ./benchmarks/dynamo/torchbench.py --accuracy --amp --amp-dtype float16 --training -d xpu -n 10 --only fbnetv3_b --backend=inductor --cold-start-latency --timeout 10800 --disable-cudagraphs
```

**Comment by chuanqi129** [comment]
dtype mismatch failures are similar with https://github.com/intel/torch-xpu-ops/issues/3290

**Comment by mengfei25** [comment]
Verified with PR 180309, all mismatched cases got passed.

**Comment by chuanqi129** [comment]
All issues are fixed
